dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java

```diff
@@ -107,8 +109,74 @@
 
     private transient ContentletAPI contentletAPI;
     private transient UserAPI userAPI;
+	private transient WriteRequest.RefreshPolicy refreshPolicy;
+
+	public Object getRefreshPolicy () {
+
+		return refreshPolicy;
+	}
+
+	/**
+	 * This method is used more a low-level library see these methods to assign the refresh policy
+	 * {@link #waitUntilContentRefresh()}
+	 * {@link #immediateContentRefresh()}
+	 *
+	 * @param refreshPolicy
+	 */
+	public void setRefreshPolicy (final Object refreshPolicy) {
+
+		if (null != refreshPolicy && refreshPolicy instanceof WriteRequest.RefreshPolicy) {
+			this.refreshPolicy = (WriteRequest.RefreshPolicy) refreshPolicy;
+		}
+	}
+	/**
+	 * If true, if the content refresh strategy is the default one.
+	 * @return Boolean
+	 */
+	public boolean isDefaultContentRefresh() {
+		return null == refreshPolicy || refreshPolicy == WriteRequest.RefreshPolicy.NONE;
+	}
+
+	/**
+	 * If true, if the content refresh strategy is the wait until content research.
+	 * @return Boolean
+	 */
+	public boolean isWaitUntilContentRefresh() {
+		return null != refreshPolicy && refreshPolicy == WriteRequest.RefreshPolicy.WAIT_UNTIL;
+	}
 
 
+	/**
+	 * If true, if the content refresh strategy is the immediate content research.
+	 * @return Boolean
+	 */
+	public boolean isImmediateContentRefresh () {
+
+		return null != refreshPolicy && this.refreshPolicy == WriteRequest.RefreshPolicy.IMMEDIATE;
+	}
+
+	/**
+	 * A process saved with this flag on, will wait until the content is already searchable in the index.
+	 */
+	public void waitUntilContentRefresh () {
+
+		this.refreshPolicy = WriteRequest.RefreshPolicy.WAIT_UNTIL;
+	}
```

Comment:
If getIndexPolicy returned an enum, we would not need all these boolean methods, right?